nuxeo-features/nuxeo-automation/nuxeo-automation-core/src/main/java/org/nuxeo/ecm/automation/core/operations/document/UpdateDocument.java

```diff
@@ -53,19 +53,17 @@
     protected String changeToken = null;
 
     @OperationMethod(collector = DocumentModelCollector.class)
-    public DocumentModel run(DocumentModel doc) throws ConflictOperationException, IOException {
+    public DocumentModel run(DocumentModel doc) throws ConcurrentUpdateException, IOException {
 
         if (changeToken != null) {
             // Check for dirty update
-            String repoToken = doc.getChangeToken();
-            if (!changeToken.equals(repoToken)) {
-                throw new ConflictOperationException(doc);
-            }
+            save = true;
```

Comment:
Actually the `save = true` was useful when a previous version of the code was catching `ConcurrentUpdateException` and rethrowing as `ConflictOperationException` but I since change the error logic to be able to deal with `ConcurrentUpdateException` directly at any point in the operation chain so I think I'll remove the `save = true`.